src/classes/TestResult.js

```diff
@@ -106,15 +121,21 @@ export default class TestResult extends BubblingEventTarget {
 				this.error = e;
 			}
 			finally {
-				await this.test.afterEach?.();
+				await test.afterEach?.();
 
 				if (!this.parent) {
 					// We are running the test in isolation, so we need to run afterAll
-					await this.test.afterAll?.();
+					await test.afterAll?.();
 				}
 			}
 		});
 
+		if (timeoutId) {
+			// If we are here, the test didn't timeout
+			// Clean up
+			clearTimeout(timeoutId);
```

Comment:
(Non-blocking nit) Do we need this check? What will happen if we just call `clearTimeout()` without checking? I think it would just be a no-op.

Reply:
You are absolutely right. There is no harm in just calling `clearTimeout()` in any case. Thanks! Fixed.